zipline/assets/asset_writer.py

```diff
@@ -268,6 +344,21 @@ def _write_futures(self, futures, bind):
     def _write_equities(self, equities, bind):
         self._write_assets(equities, self.equities, 'equity', bind)
 
+    def check_for_tables(self, engine):
+        """
+        Checks if any tables are present in the current assets database.
+
+        Returns
+        -------
+        bool
+            True if any tables are present, otherwise False.
+        """
+        conn = engine.connect()
+        for table_name in table_names:
```

Comment:
In the case of a schema change, table_names[0] may not exist anymore but we'd want this method to return True. The only way I could think of to guarantee this is by checking all table names. It will return after the first table_name, anyway, which would mean that in the nominal case it is identical in execution to
```
return engine.dialect.has_table(conn, table_names[0])